src/Rules/Install.hs

```diff
@@ -42,41 +56,61 @@ installLibExecs = do
   libexecDir <- (-/- "bin") <$> ghclibdir
   dest <- destDir
   installDir (dest ++ libexecDir)
-  let programContexts = liftM2 programContext [minBound..] knownPackages  
-  forM_ programContexts $ \context@Context{..} -> do
-      activePackages <- interpretInContext context getPackages
-      when (stage == Stage0 || package == ghc) $ do
-          when (not (isLibrary package) && package `elem` activePackages) $ do
-              let bin = programInplaceLibPath -/- programName context <.> exe
-              need [bin]
-              installProgram bin (dest ++ libexecDir)
+  forM_ installStages $ \(pkg, mStgs) -> do
+    stgs <- mStgs
+    forM_ stgs $ \stg -> do
+      let context = programContext stg pkg
+      let bin = programInplaceLibPath -/- programName context <.> exe
+      need [bin]
+      installProgram bin (dest ++ libexecDir)
+      when (pkg == ghc) $ do
+        moveFile (dest ++ libexecDir -/- programName context <.> exe)
+                 (dest ++ libexecDir -/- "ghc" <.> exe)
+
+ghcSplitPkg :: Package
+ghcSplitPkg = utility "ghc-split"
+
+-- TODO: Consider Stage1Only
+installStages :: [(Package, Action [Stage])]
+installStages =
+  [ (ghc, maybeToList <$> latestBuildStage ghc)
+  , (ghcPkg, return [ Stage1 ])
+  , (ghcSplitPkg, return [ Stage0 ])
+  , (hp2ps, return [ Stage1 ])
+  , (hpc, return [ Stage1 ])
+  , (hsc2hs, return [ Stage1 ])
+  , (runGhc, return [ Stage1 ])
+  , (unlit, return [ Stage1 ]) ]
```

Comment:
And `latestBuildStage` will return exactly `Stage1`, right? 

The `latestBuildStage` was my attempt to specify the intent: we install packages that are built by the latest (most feature-rich) stage of GHC we have.

Can you give an example where using `latestBuildStage` instead of a hard-coded stage is wrong?